src/lib/y2storage/boot_requirements_checker.rb

```diff
@@ -61,26 +67,36 @@ def needed_partitions(target = :desired)
 
   protected
 
-    attr_reader :settings
+    # @return [Devicegraph] starting situation.
     attr_reader :devicegraph
 
+    # @return [Array<Planned::Device>] devices that are already planned to be
+    #   added to the starting devicegraph.
+    attr_reader :planned_devices
+
+    # @return [String, nil] device name of the disk that the system will try to
+    #   boot first. Only useful in some scenarios like legacy boot.
+    attr_reader :boot_disk_name
+
     def arch
       @arch ||= StorageManager.instance.arch
     end
 
     def strategy
       return @strategy unless @strategy.nil?
 
-      if arch.x86? && arch.efiboot?
-        @strategy = BootRequirementsStrategies::UEFI.new(settings, devicegraph)
-      elsif arch.s390?
-        @strategy = BootRequirementsStrategies::ZIPL.new(settings, devicegraph)
-      elsif arch.ppc?
-        @strategy = BootRequirementsStrategies::PReP.new(settings, devicegraph)
-      end
-
-      # Fallback to Legacy as default
-      @strategy ||= BootRequirementsStrategies::Legacy.new(settings, devicegraph)
+      klass =
+        if arch.x86? && arch.efiboot?
+          BootRequirementsStrategies::UEFI
+        elsif arch.s390?
+          BootRequirementsStrategies::ZIPL
+        elsif arch.ppc?
+          BootRequirementsStrategies::PReP
```

Comment:
well, then consider this code confusing :) having strategy for prep that do not require prep is confusing. What about calling this strategy PPC strategy? Because you actually do not require prep, but require booting strategy for ppc include powernv